pytorch_lightning/trainer/trainer.py

```diff
@@ -101,8 +102,8 @@ def __init__(
         check_val_every_n_epoch: int = 1,
         fast_dev_run: Union[int, bool] = False,
         accumulate_grad_batches: Union[int, Dict[int, int], List[list]] = 1,
-        max_epochs: int = 1000,
-        min_epochs: int = 1,
+        max_epochs: Optional[int] = None,
+        min_epochs: Optional[int] = None,
```

Comment:
this is confusing, you say there is a default number but you make it as None
in general None default is used for deprecated or mutable values (such as list)

Reply:
that's fair. if I set a default for min/max_epochs, then users who want to do training based purely on steps need to specify this:
```
trainer = Trainer(min_steps=M, max_steps=N, min_epochs=None, max_epochs=None, ...)
```

instead they can simply specify this:
```
trainer = Trainer(min_steps=M, max_steps=N, ...)
```

so by setting these defaults to `None` in the constructor and handling the initialization elsewhere, we save users some of the boilerplate code from having to null out the epoch arguments if they don't use them. what do you think?